carbon-design-system/carbon
# fix(Loading): improve screenreader support

Refs https://github.com/carbon-design-system/carbon/issues/11259

Work to improve screen reader support with `Loading` and `InlineLoading`

#### Changelog

**Changed**
- Reduced default descriptive text to "Loading"
- With `InlineLoading`, if an `iconDescription` is not provided, it will use the status as the `title` text. 
- If text is provided to `InlineLoading` via `description`, this will be read by the screenreader. If no text is provided, the `title` text is picked up (either the `iconDescription` if provided, or the `status`)

**Removed**
- The `label` element has been removed, as it was adding to the noise for screenreaders.
- `aria-labelledby` has been removed, since the `label` it was referring to no longer exists
- The `id` prop is deprecated as it is no longer used.

#### Testing / Reviewing

Open up the stories for `Loading` and `InlineLoading`. 

- Use the Accessibility checker and ensure there are no violations. 
-  Try to remove/change the `iconDescription` and `description` props for `InlineLoading` and test the changes with the MacOS VoiceOver tool.
- Try to remove/change the `description` prop for `Loading` and test the changes with the MacOS VoiceOver tool.
